python/lsst/jointcal/jointcal.py

```diff
@@ -614,6 +609,67 @@ def _do_load_refcat_and_fit(self, associations, defaultFilter, center, radius,
 
         return result
 
+    def _load_reference_catalog(self, refObjLoader, center, radius, filterName,
+                                applyColorterms=False):
+        """Load the necessary reference catalog sources, convert fluxes to
+        correct units, and apply color term corrections if requested.
+
+        Parameters
+        ----------
+        refObjLoader : `lsst.meas.algorithms.LoadReferenceObjectsTask`
+            The reference catalog loader to use to get the data.
+        center : `lsst.geom.SpherePoint`
+            The center around which to load sources.
+        radius : `lsst.geom.Angle`
+            The radius around ``center`` to load sources in.
+        filterName : `str`
+            The name of the camera filter to load fluxes for.
+        applyColorterms : `bool`
+            Apply colorterm corrections to the refcat for ``filterName``?
+
+        Returns
+        -------
+        refCat : `lsst.afw.table.SimpleCatalog`
+            The loaded reference catalog.
+        fluxField : `str`
+            The name of the reference catalog flux field appropriate for ``filterName``.
+        """
+        skyCircle = refObjLoader.loadSkyCircle(center,
+                                               afwGeom.Angle(radius, afwGeom.radians),
+                                               filterName)
+
+        # Need memory contiguity to get reference filters as a vector.
+        if not skyCircle.refCat.isContiguous():
+            refCat = skyCircle.refCat.copy(deep=True)
```

Comment:
Okay, here's a continuity check, but does there need to be one in `pipe_tasks` as well?

Reply:
Probably. That would be ticket DM-14061.